src/pamiq_core/console/cui.py

```diff
@@ -1,84 +1,109 @@
 import argparse
-import cmd
 import json
-from typing import override
 
 import httpx
+from prompt_toolkit import prompt
+from prompt_toolkit.completion import WordCompleter
 
 
-class Console(cmd.Cmd):
+class Console:
     """pamiq-console.
 
     Users can Control pamiq with CUI interface interactively.
     """
 
-    intro = 'Welcome to the PAMIQ console. "help" lists commands.\n'
-    prompt: str
+    status: str
 
     def __init__(self, host: str, port: int) -> None:
         """Initialize CUI interface."""
         super().__init__()
         self._host = host
         self._port = port
-        self._fetch_status()
+        self._completer = WordCompleter(self.get_all_commands())
+        self.fetch_status()
 
-    @override
-    def onecmd(self, line: str) -> bool:
+    def fetch_status(self) -> None:
+        """Check WebAPI status."""
+        try:
+            response = httpx.get(f"http://{self._host}:{self._port}/api/status")
+        except httpx.RequestError:
+            self.status = "offline"
+            return
+        self.status = json.loads(response.text)["status"]
+
+    def get_all_commands(self) -> list[str]:
+        return [
+            attr[len("command_") :] for attr in dir(self) if attr.startswith("command_")
+        ]
+
+    def run_command(self, command: str) -> bool | None:
         """Check connection status before command execution."""
-        # Update status when every command execution.
-        status = self._fetch_status()
+        # Update self.status before command execution.
+        self.fetch_status()
         # Check command depend on WebAPI
-        cmd_name, _, _ = self.parseline(line)
-        if cmd_name in ["pause", "p", "resume", "r", "save", "shutdown"]:
+        if command in ["pause", "p", "resume", "r", "save", "shutdown"]:
             # Check if WebAPI available.
-            if status == "offline":
-                print(f'Command "{cmd_name}" not executed. Can\'t connect AMI system.')
+            if self.status == "offline":
+                print(f'Command "{command}" not executed. Can\'t connect AMI system.')
                 return False
         # Execute command
-        return super().onecmd(line)
-
-    def get_all_commands(self) -> list[str]:
-        return [attr[3:] for attr in dir(self) if attr.startswith("do_")]
+        loop_end = getattr(self, f"command_{command}")()
+        # Update self.status after command execution.
+        self.fetch_status()
+        # If True, main_loop ends.
+        return loop_end
+
+    def main_loop(self) -> None:
+        """Running CUI interface."""
+        print('Welcome to the PAMIQ console. "help" lists commands.')
```

Comment:
改行を二つ入れておくと見やすくなると思います。
```suggestion
        print('Welcome to the PAMIQ console. "help" lists commands.\n\n')
```

Reply:
https://github.com/MLShukai/pamiq-core/pull/217/commits/b1ef2a0dc7c58cef5b9faa86e98e6dbe8aece65d 2つだと逆に見かけの空白部分が大きくなりすぎるから1つにしといた